packages/react/src/functions/safe-dynamic-require.ts

```diff
@@ -26,10 +26,12 @@ export let safeDynamicRequire: typeof require;
  * In Summary:
  *
  * 1. Node -> globalThis.require does not work
- * 2. Client -> typeof require === 'function' does not work because of overoptimization by the compiler
- * 3. Cloudflare edge -> only globalThis.require works
+ * 2. Cloudflare edge -> only globalThis.require works
  */
-if (typeof globalThis.require === 'function' || Builder.isServer) {
+if (
+  typeof globalThis.require === 'function' ||
```

Comment:
Do we need to change this to 
```suggestion
  typeof globalThis?.require === 'function' ||
```
in case `globalThis` is undefined?